app/src/main/java/org/simple/clinic/scanid/QRCodeJsonParser.kt

```diff
@@ -0,0 +1,14 @@
+package org.simple.clinic.scanid
+
+import com.squareup.moshi.Moshi
+import javax.inject.Inject
+
+class QRCodeJsonParser @Inject constructor(
+    private val moshi: Moshi
+) {
+
+  fun parseQRCodeJson(text: String): IndiaNHIDInfoPayload? {
+    val adapter = moshi.adapter(IndiaNHIDInfoPayload::class.java)
```

Comment:
This adapter can be a class property instead of being fetched every time the method is called, right?

Reply:
Makes sense but If it were to fetch every time what would the consequences be?

Reply:
It will still continue to work, it's just we end up creating a new adapter instance every time this method is called and not that memory efficient, since we can reuse an already available instance, we don't want to create it multiple times.